core/types/cx_receipt.go

```diff
@@ -72,3 +76,106 @@ type CXMerkleProof struct {
 	ShardIDs      []uint32      // order list, records destination shardID
 	CXShardHashes []common.Hash // ordered hash list, each hash corresponds to one destination shard's receipts root hash
 }
+
+// CXReceiptsProof carrys the cross shard receipts and merkle proof
+type CXReceiptsProof struct {
+	Receipts    CXReceipts
+	MerkleProof *CXMerkleProof
```

Comment:
Why CXMerkleProof is pointer while the CXReceipts is not?

Reply:
CXReceipts is already an array:  []*CXReceipts, so we don't use another pointer here